### ISSUE TITLE
Build failing with "multiple definition" errors

### ISSUE THREAD

**Issue by aJunk** [OLSR/OONF#28]
Hi,

For me both the master and development branch fail due to some symbols being defined more than once. I think this is due to some variables not being declared extern in their respective header files.

Namely, those are: 
NETADDR_SOCKET_IPV4_ANY, NETADDR_SOCKET_IPV6_ANY, LOG_DLEP_RADIO, LOG_NL80211, LOG_MPR, LOG_NHDP, LOG_NHDP_R, LOG_NHDP_W, LOG_OLSRV2, LOG_OLSRV2_R, LOG_OLSRV2_ROUTING, LOG_OLSRV2_W

If this is actually a bug and not my setup being wrong I would love to submit a pull request for the fix.

**Comment by HRogge** [comment]
I think it might be a good thing for you say what you used as your build chain and how you generate the error.

Did you use a different compiler than GCC for example (I think I never tested anything else).

**Comment by aJunk** [comment]
Oh, yes, of course, I am sorry! I completely skipped that info.

I am building on Arch Linux with GCC 10.1.0.
I followed the instructions in the Makefile and ran `cmake ..` in the build directory and invoked `make` after that.
I just tried it with 9.2.1 on a different Machine and 8.0 on the same machine which both work.

**I think I have identified the problem:** In GCC versions before 10, symbol conflicts caused by ommiting the `extern` keyword in declarations in header files were silently ignored. GCC 10 defaults to '-fno-common' which raises an error.
This is documented [here](https://gcc.gnu.org/gcc-10/porting_to.html).

The behaviour can be verified by using GCC <10 and adding '-fno-common' to the linker flags in the CMakeCache.txt file prior to running `make`.

**Comment by HRogge** [comment]
Thank you for the quick diagnosis of the problem... if you already have a solution feel free to send me a push request to the develop branch... if not I will look into it at the weekend.

**Comment by aJunk** [comment]
I would be very glad to do so!
Although I have never made a PR before, so please let me know if I do something wrong.

**Comment by aJunk** [comment]
I created a pull request for this issue.

However, when merging the changes into the development branch another problem cropped up with the DNS library. I have created a new issue, I hope that is alright.

**Comment by HRogge** [comment]
Yes, the DNS library was a pain in the ass... I will see what I can do, most likely add more special GCC Arguments for its compilation.

**Comment by aJunk** [comment]
I documented my findings in issue #30, it took me a while since I ran out of disk space half way through...
I do not think this can be worked around that way, since apparently this should not have worked to begin with and was allowed through a bug in GCC.

**Comment by jpo-github-work** [comment]
The problem is still not fixed.

**Comment by jpo-github-work** [comment]
This should be fixed now with #40 and #42 merged.

**Comment by mathiashro** [comment]
Thank you & closing.
